core/rq-worker/file_watch.py

```diff
@@ -108,124 +111,81 @@ def template_queue(path, base_dir="/var/lib/docker/data/"):
-                        pass
-            if len(core_containers) > 0:
-                os.system('python2.7 '+vent_dir+'template_parser.py core start')
-            # !! TODO failing, so commenting out for now
-            #c.restart(current_container)
-        elif template == "core.template":
-            # check if the container is this one and kill it last!
-            hostname = os.environ.get('HOSTNAME')
-            current_container = hostname
-            # check if active/passive are disabled, and kill/remove containers
-            if os.path.isfile(template_dir+'core.template'):
-                config.read(template_dir+'core.template')
-            local_collection = []
-            passive = False
-            active = False
-            if config.has_section("local-collection") and config.options("local-collection"):
-                local_collection = config.options("local-collection")
-            for collector in local_collection:
-                val = config.get("local-collection", collector)
-                if collector == "passive" and val == "on":
-                    passive = True
-                if collector == "active" and val == "on":
-                    active = True
-            if not passive:
-                passive_containers = c.containers(all=True, filters={'name':"passive"})
-                for cont in passive_containers:
+        # remove disabled running containers
+        for container in containers:
+            for name in container["Names"]:
+                if name in disabled:
                     try:
-                        c.kill(cont['Id'])
-                        c.remove_container(cont['Id'])
-                    except Exception as e:
-                        pass
-            if not active:
-                active_containers = c.containers(all=True, filters={'name':"active"})
-                for cont in active_containers:
-                    try:
-                        c.kill(cont['Id'])
-                        c.remove_container(cont['Id'])
+                        c.kill(container["Id"])
+                        c.remove_container(container["Id"])
+                        continue
                     except Exception as e:
                         pass
 
-            # kill and remove core containers
-            core_containers = c.containers(quiet=True, all=True, filters={'name':"core"})
-            for cont in core_containers:
-                if cont['Id'].startswith(current_container):
-                    current_container = cont['Id']
+        core_containers = c.containers(all=True, filters={'name':"core-"})
+
+        # remove core containers, so when restarted, behavior matches current configuration
+        for container in core_containers:
+            try:
+                if container["Status"] == "exited":
+                    c.remove_container(container["Id"])
+                elif container["Name"].startswith(os.environ.get('HOSTNAME')):
```

Comment:
I think you want `container['Id']` here, since the hostname is set to the ID of the container, I believe?